include/libassert/platform.hpp

```diff
@@ -226,24 +226,32 @@ namespace libassert::detail {
     }
 }
 
-#if LIBASSERT_IS_CLANG || LIBASSERT_IS_GCC
- #if LIBASSERT_IS_GCC
-  #define LIBASSERT_WARNING_PRAGMA_PUSH_GCC _Pragma("GCC diagnostic push")
-  #define LIBASSERT_WARNING_PRAGMA_POP_GCC _Pragma("GCC diagnostic pop")
-  #define LIBASSERT_WARNING_PRAGMA_PUSH_CLANG
-  #define LIBASSERT_WARNING_PRAGMA_POP_CLANG
- #else
-  #define LIBASSERT_WARNING_PRAGMA_PUSH_GCC
-  #define LIBASSERT_WARNING_PRAGMA_POP_GCC
-  #define LIBASSERT_WARNING_PRAGMA_PUSH_CLANG _Pragma("GCC diagnostic push")
-  #define LIBASSERT_WARNING_PRAGMA_POP_CLANG _Pragma("GCC diagnostic pop")
- #endif
+#if LIBASSERT_IS_CLANG
+ #define LIBASSERT_PRAGMA_CLANG(x) _Pragma(#x)
+#else
+ #define LIBASSERT_PRAGMA_CLANG(x)
+#endif
+#define LIBASSERT_WARNING_PRAGMA_PUSH_CLANG LIBASSERT_PRAGMA_CLANG(GCC diagnostic push)
+#define LIBASSERT_WARNING_PRAGMA_POP_CLANG LIBASSERT_PRAGMA_CLANG(GCC diagnostic pop)
+#define LIBASSERT_WARNING_PRAGMA_IGNORED_CLANG(x) LIBASSERT_PRAGMA_CLANG(GCC diagnostic ignored #x)
+
+#if LIBASSERT_IS_GCC
+ #define LIBASSERT_PRAGMA_GCC(x) _Pragma(#x)
+#else
+ #define LIBASSERT_PRAGMA_GCC(x)
+#endif
+#define LIBASSERT_WARNING_PRAGMA_PUSH_GCC LIBASSERT_PRAGMA_GCC(GCC diagnostic push)
+#define LIBASSERT_WARNING_PRAGMA_POP_GCC LIBASSERT_PRAGMA_GCC(GCC diagnostic pop)
+#define LIBASSERT_WARNING_PRAGMA_IGNORED_GCC(x) LIBASSERT_PRAGMA_GCC(GCC diagnostic ignored #x)
+
+#if LIBASSERT_IS_MSVC
+ #define LIBASSERT_PRAGMA_MSVC(x) __pragma(x)
```

Comment:
MSVC2022 today also supports the standard `_Pragma(string)`: https://learn.microsoft.com/en-us/cpp/preprocessor/pragma-directives-and-the-pragma-keyword?view=msvc-170